docs/en/sql-reference/functions/type-conversion-functions.md

```diff
@@ -49,105 +49,1378 @@ SETTINGS cast_keep_nullable = 1
 └──────────────────┴─────────────────────┴──────────────────┘
 ```
 
-## toInt(8\|16\|32\|64\|128\|256)
+## toInt8
 
-Converts an input value to a value the [Int](../data-types/int-uint.md) data type. This function family includes:
+Converts an input value to a value of type [`Int8`](../data-types/int-uint.md). Throws an exception in case of an error.
 
-- `toInt8(expr)` — Converts to a value of data type `Int8`.
-- `toInt16(expr)` — Converts to a value of data type `Int16`.
-- `toInt32(expr)` — Converts to a value of data type `Int32`.
-- `toInt64(expr)` — Converts to a value of data type `Int64`.
-- `toInt128(expr)` — Converts to a value of data type `Int128`.
-- `toInt256(expr)` — Converts to a value of data type `Int256`.
+**Syntax**
+
+```sql
+toInt8(expr)
+```
+
+**Arguments**
+
+- `expr` — Expression returning a number or a string representation of a number. [Expression](../syntax.md/#syntax-expressions).
+
+Supported types:
+- (U)Int8/16/32/64/128/256
+- Float*
+- String representations of (U)Int8/16/32/128/256
+
+Unsupported types:
+- Float values `NaN` and `Inf` throw an exception.
+- String representations of binary and hexadecimal values, e.g. `SELECT toInt8('0xc0fe');`
+
+:::note
+If the input value cannot be represented within the bounds of [Int8](../data-types/int-uint.md), the result over or under flows. This is not considered an error.  
+For example: `SELECT toInt8(128) == -128;`, `SELECT toInt8(128.0) == -128;`,  `SELECT toInt8('128') == -128;`.
+:::
+
+**Returned value**
+
+- 8-bit integer value. [Int8](../data-types/int-uint.md).
+
+:::note
+The function uses [rounding towards zero](https://en.wikipedia.org/wiki/Rounding#Rounding_towards_zero), meaning it truncates fractional digits of numbers.
+:::
+
+**Example**
+
+Query:
+
+```sql
+SELECT
+    toInt8(-8),
+    toInt8(-8.8),
+    toInt8('-8');
+```
+
+Result:
+
+```response
+   ┌─toInt8(-8)─┬─toInt8(-8.8)─┬─toInt8('-8')─┐
+1. │         -8 │           -8 │           -8 │
+   └────────────┴──────────────┴──────────────┘
+```
+
+**See also**
+
+- [`toInt8OrZero`](#toint8orzero).
+- [`toInt8OrNull`](#toint8ornull).
+- [`toInt8OrDefault`](#toint8ordefault).
+
+## toInt8OrZero
```

Comment:
(side note, if someone asked me, I did not know that `toInt8OrZero` supports only string arguments. That's actually surprising ... anyways, let's document it as is)